File: LeetCode/0064. Minimum Path Sum/Solution.py
class Solution:
    def minPathSum(self, grid: List[List[int]]) -> int:
    
        for i in range(len(grid)):
            
            for j in range(len(grid[0])):

                if i > 0 or j > 0:
                    grid[i][j] += min(
                        grid[i-1][j] if i > 0 else float("inf"),
                        grid[i][j-1] if j > 0 else float("inf")
                    )

        return grid[-1][-1]

File: LeetCode/0064. Minimum Path Sum/Solution.cpp
class Solution {
public:
    int minPathSum(vector<vector<int>>& grid) {
        
        for (int i = 0; i < grid.size(); i++)
        {
            for (int j = 0; j < grid[0].size(); j++)
            {   
                if (i > 0 || j > 0)
                {
                    grid[i][j] += min(i > 0 ? grid[i-1][j] : 201, j > 0 ? grid[i][j-1] : 201);
                }

            }
            
        }
        
        return grid.back().back();
    }
};
